### ISSUE TITLE
Bumblebee not working on MSI/GT 950M nvidia

### ISSUE THREAD

**Issue by EgamoPi** [Bumblebee-Project/Bumblebee#862]
Good morning, 
I installed bumblebee and configured it, following this tutorial quit fresh : https://fixmynix.com/install-and-configure-nvidia-optimus-with-bumblebee-in-debian/ . I still have problems : while running "optirun -vv glxgears" i have the following errors : 
omega@egamo:~$ sudo optirun -vv glxgears
[sudo] password for omega: 
[ 1993.439907] [DEBUG]Reading file: /etc/bumblebee/bumblebee.conf
[ 1993.440115] [INFO]Configured driver: nvidia
[ 1993.440381] [DEBUG]optirun version 3.2.1 starting...
[ 1993.440411] [DEBUG]Active configuration:
[ 1993.440416] [DEBUG] bumblebeed config file: /etc/bumblebee/bumblebee.conf
[ 1993.440421] [DEBUG] X display: :8
[ 1993.440426] [DEBUG] LD_LIBRARY_PATH: /usr/lib/x86_64-linux-gnu/nvidia:/usr/lib/i386-linux-gnu/nvidia:/usr/lib/nvidia
[ 1993.440431] [DEBUG] Socket path: /var/run/bumblebee.socket
[ 1993.440436] [DEBUG] Accel/display bridge: auto
[ 1993.440441] [DEBUG] VGL Compression: proxy
[ 1993.440448] [DEBUG] VGLrun extra options: 
[ 1993.440453] [DEBUG] Primus LD Path: /usr/lib/x86_64-linux-gnu/primus:/usr/lib/i386-linux-gnu/primus:/usr/lib/primus:/usr/lib32/primus
[ 1993.440515] [DEBUG]Using auto-detected bridge virtualgl
[ 1995.656633] [INFO]Response: No - error: [XORG] (EE) NVIDIA(GPU-0): Failed to initialize the NVIDIA GPU at PCI:1:0:0.  Please

[ 1995.656643] [ERROR]Cannot access secondary GPU - error: [XORG] (EE) NVIDIA(GPU-0): Failed to initialize the NVIDIA GPU at PCI:1:0:0.  Please

[ 1995.656645] [DEBUG]Socket closed.
[ 1995.656653] [ERROR]Aborting because fallback start is disabled.
[ 1995.656656] [DEBUG]Killing all remaining processes.
 
Any idea ?
Also if it can help my when I run optirun i can see that i am on the nvidia card and not on the intel because of the led indication on my laptop. My bbswitch is working, because I can switch back to the intel by restarting the bumblebee service. It's not a big problem, but now that i work with a monitor for my job, i need my laptop to recognize my screen (yes because my screen is not recognize) in parameters -> Display it's written unkown display and by consequence i can even use my freshly monitor white and so nice that i just bought^^ 
And by the way, sorry for my english but I hope that you will understand !

Oh and i'm on debian jessie 

and lspci command return this (if you want more info) : 
00:00.0 Host bridge: Intel Corporation Device 1910 (rev 07)
00:01.0 PCI bridge: Intel Corporation Device 1901 (rev 07)
00:02.0 VGA compatible controller: Intel Corporation Device 191b (rev 06)
00:14.0 USB controller: Intel Corporation Device a12f (rev 31)
00:14.2 Signal processing controller: Intel Corporation Device a131 (rev 31)
00:16.0 Communication controller: Intel Corporation Device a13a (rev 31)
00:17.0 SATA controller: Intel Corporation Device a103 (rev 31)
00:1c.0 PCI bridge: Intel Corporation Device a110 (rev f1)
00:1c.2 PCI bridge: Intel Corporation Device a112 (rev f1)
00:1c.3 PCI bridge: Intel Corporation Device a113 (rev f1)
00:1c.4 PCI bridge: Intel Corporation Device a114 (rev f1)
00:1d.0 PCI bridge: Intel Corporation Device a118 (rev f1)
00:1f.0 ISA bridge: Intel Corporation Device a14e (rev 31)
00:1f.2 Memory controller: Intel Corporation Device a121 (rev 31)
00:1f.3 Audio device: Intel Corporation Device a170 (rev 31)
00:1f.4 SMBus: Intel Corporation Device a123 (rev 31)
01:00.0 3D controller: NVIDIA Corporation Device 139b (rev ff)
02:00.0 Network controller: Qualcomm Atheros Device 003e (rev 20)
03:00.0 Unassigned class [ff00]: Realtek Semiconductor Co., Ltd. RTS5249 PCI Express Card Reader (rev 01)
04:00.0 Ethernet controller: Qualcomm Atheros Device e0a1 (rev 10)
3e:00.0 Non-Volatile memory controller: Toshiba America Info Systems Device 010f (rev 01)
 Thanks !
Jonathan

**Comment by bluca** [comment]
If you followed that link to the letter and you are on jessie, it will be a problem, as it suggests to add the repositories for stretch, which is very wrong. Is that the case? Or did you adapt it for jessie?

**Comment by EgamoPi** [comment]
Ouuppss... I didn't adapt ! So should I restart all the configuration ? or ? Thanks for your fast reply

**Comment by bluca** [comment]
Mixing repositories from different Debian (or other distros) is a problem https://wiki.debian.org/DontBreakDebian

You should first of all figure out what was installed from stretch, and downgrade to jessie, to avoid issues. If you don't have the terminal open anymore, you can consult /var/log/apt/history.log

Then install from backports.

**Comment by EgamoPi** [comment]
My bad, by reading back the tutorial i realized that I didn't even add the repository because i thought that i already had it : 
This is my source list, isn't sufficient ?

 #deb cdrom:[Debian GNU/Linux 8.7.1 _Jessie_ - Official amd64 DVD Binary-1 20170116-11:01]/ jessie contrib main

#deb cdrom:[Debian GNU/Linux 8.7.1 _Jessie_ - Official amd64 DVD Binary-1 20170116-11:01]/ jessie contrib main

deb http://ftp.ru.debian.org/debian/ jessie main contrib non-free
deb-src http://ftp.ru.debian.org/debian/ jessie main contrib non-free

deb http://security.debian.org/ jessie/updates main contrib non-free
deb-src http://security.debian.org/ jessie/updates main contrib non-free

# jessie-updates, previously known as 'volatile'
deb http://ftp.ru.debian.org/debian/ jessie-updates main contrib non-free
deb-src http://ftp.ru.debian.org/debian/ jessie-updates main contrib non-free

or should i still add : deb http://ftp.debian.org/debian/ jessie main contrib non-free ?

**Comment by bluca** [comment]
That's good news, as recovering it's very painful when these mixup happens.

Add this line:

```
deb http://ftp.ru.debian.org/debian/ jessie-backports main contrib non-free
```

Then run ```sudo apt-get update; sudo apt-get install -t jessie-backports bumblebee-nvidia nvidia-driver```

**Comment by EgamoPi** [comment]
Ok so i just did that, installation worked but new problem : 
omega@egamo:~$ sudo optirun -vv glxgears
[sudo] password for omega: 
[  131.028125] [DEBUG]Reading file: /etc/bumblebee/bumblebee.conf
[  131.028630] [DEBUG]optirun version 3.2.1 starting...
[  131.028645] [DEBUG]Active configuration:
[  131.028650] [DEBUG] bumblebeed config file: /etc/bumblebee/bumblebee.conf
[  131.028655] [DEBUG] X display: :8
[  131.028660] [DEBUG] LD_LIBRARY_PATH: 
[  131.028665] [DEBUG] Socket path: /var/run/bumblebee.socket
[  131.028671] [DEBUG] Accel/display bridge: auto
[  131.028676] [DEBUG] VGL Compression: proxy
[  131.028682] [DEBUG] VGLrun extra options: 
[  131.028687] [DEBUG] Primus LD Path: /usr/lib/x86_64-linux-gnu/primus:/usr/lib/i386-linux-gnu/primus:/usr/lib/primus:/usr/lib32/primus
[  131.045055] [DEBUG]Using auto-detected bridge virtualgl
[  131.850628] [INFO]Response: No - error: [XORG] (EE) [drm] KMS not enabled

[  131.850638] [ERROR]Cannot access secondary GPU - error: [XORG] (EE) [drm] KMS not enabled

[  131.850641] [DEBUG]Socket closed.
[  131.850664] [ERROR]Aborting because fallback start is disabled.
[  131.850667] [DEBUG]Killing all remaining processes.

and when i want to restart bumblebee to come back on the intel card : 
omega@egamo:~$ sudo service bumbleed restart
Failed to restart bumbleed.service: Unit bumbleed.service failed to load: No such file or directory.

i tried to google the first error but all the answers are to continue using nouveau driver :( and also i still have an unknown display, any idea ?

**Comment by bluca** [comment]
Have you rebooted?

**Comment by EgamoPi** [comment]
Yes I have rebooted and the output is : 
sudo dkms status
bbswitch, 0.8, 3.16.0-4-amd64, x86_64: installed
nvidia-current, 375.26, 3.16.0-4-amd64, x86_64: installed

**Comment by bluca** [comment]
What is the provider for glx? Run:

```
sudo update-glx --config glx
```

The selected one should be ```/usr/lib/nvidia/bumblebee```

**Comment by EgamoPi** [comment]
omega@egamo:~$ sudo update-glx --config glx
[sudo] password for omega: 
Il existe 3 choix pour l'alternative glx (qui fournit /usr/lib/glx).

  Sélection   Chemin                     Priorité  État
------------------------------------------------------------
  0            /usr/lib/nvidia/bumblebee   125       mode automatique
* 1            /usr/lib/mesa-diverted      5         mode manuel
  2            /usr/lib/nvidia             100       mode manuel
  3            /usr/lib/nvidia/bumblebee   125       mode manuel

Appuyez sur <Entrée> pour conserver la valeur par défaut[*] ou choisissez le numéro sélectionné :
Traitement des actions différées (« triggers ») pour glx-alternative-mesa (0.7.3~bpo8+1) ...
Traitement des actions différées (« triggers ») pour libc-bin (2.19-18+deb8u7) ...

So mesa-diverted is selected, i have two choices for bumblebee, which one should I use automatic mode or manuel mode ?

**Comment by bluca** [comment]
automatic

**Comment by EgamoPi** [comment]
IT WORKS !!!!! Thank you so much !! Last question and promise i'm no more disturbing you : The display still shows unknown, do you have any idea why ?

**Comment by bluca** [comment]
If it's the laptop display you are referring to, then it's not wired to the nvidia card, so it is not related to bumblebee.

If it's a newish laptop (and with a 950m it must be) then it might be because the intel driver is too old.

Try:

```sudo apt-get install -t jessie-backports xserver-xorg-video-intel```

To get a newer version.

Since the bumblebee issue is solved I'll close this for now.

**Comment by EgamoPi** [comment]
I'm sorry to reopen this topic, but I just did a fresh installation on the same computer but this time with stretch, I followed the same explaination and the same solution that you offer everything seem to be working. The "dkms status" and "update glx" returns are good. I used also the backports of stretch after trying without and yea. It just freeze after login into my session, and the nomodeset doesn't work.
